### PR TITLE
feat(next-queries): add `NextQueriesList` component.

### PR DESCRIPTION
**Pull request by javieri-empathy**
# Description

Adds the new `NextQueriesList` component. This component allows to insert next queries in an injected list of `SearchItems`. How this component inserts the next queries can be configured using the props.

# Context

This is needed to support an awesome grid of search items.

# Test

Check use cases, documentation, that the components are properly exported...

### CODE REVIEW

**Review by tajespasarela — CHANGES_REQUESTED** [review]

File: packages/x-components/src/x-modules/search/components/banners-list.vue [hunk 1]
```diff
@@ -124,13 +125,14 @@ _Type any term in the input field to try it out!_
 ```vue
 <template>
   <div>
-    <SearchInput />
+    <SearchInput :instant="true" />
```
[the remark this answers is not in the document]
Comment (javieri-empathy) [reply packages/x-components/src/x-modules/search/components/banners-list.vue]: I don't know why I thought it was `false`, must have checked the prop. I reverted the changes

File: packages/x-components/src/components/items-list.vue [hunk 2]
```diff
@@ -1,41 +1,41 @@
 <template>
   <component
     :is="animation"
-    v-if="searchItems.length"
+    v-if="items.length"
     tag="ul"
     class="x-search-items-list"
```
Comment (tajespasarela) [inline packages/x-components/src/components/items-list.vue:6]: rename also the CSS classes and the `data-test` attribute

File: packages/x-components/src/components/__tests__/items-list.spec.ts [hunk 3]
```diff
@@ -29,22 +29,22 @@ function renderSearchItemsList({
...
 
-  it("doesn't render the list if `searchItems` are provided", () => {
-    const { wrapper } = renderSearchItemsList({
-      searchItems: []
+  it("doesn't render the list if `items` are provided", () => {
+    const { wrapper } = renderItemsList({
+      items: []
     });
 
     expect(wrapper.find(getDataTestSelector('search-items-list')).exists()).toBe(false);
```
Comment (tajespasarela) [inline packages/x-components/src/components/__tests__/items-list.spec.ts:42]: rename the data test selector

File: packages/x-components/src/components/__tests__/items-list.spec.ts [hunk 4]
```diff
@@ -29,22 +29,22 @@ function renderSearchItemsList({
...
   });
 
   it('renders the items list passed as property', () => {
-    const { wrapper } = renderSearchItemsList({
-      searchItems: resultsStub
+    const { wrapper } = renderItemsList({
+      items: resultsStub
     });
 
     const itemsWrapperArray = wrapper.findAll('.x-search-items-list__item');
```
Comment (tajespasarela) [inline packages/x-components/src/components/__tests__/items-list.spec.ts:50]: rename the data test selector

**Review by tajespasarela — CHANGES_REQUESTED** [review]
add the BREAKING CHANGES to the PR description

File: packages/x-components/src/components/items-list.vue [hunk 5]
```diff
@@ -50,7 +50,7 @@
      * @public
```
Comment (tajespasarela) [inline packages/x-components/src/components/items-list.vue:50]: Rename de doc above this line

File: packages/x-components/src/views/Layout.vue [hunk 6]
```diff
@@ -194,29 +194,31 @@
...
+                      <Banner :banner="banner" />
+                    </template>
 
-                  <template #Promoted="{ item: promoted }">
-                    <Promoted :promoted="promoted" />
-                  </template>
-                </BaseVariableColumnGrid>
+                    <template #Promoted="{ item: promoted }">
+                      <Promoted :promoted="promoted" />
+                    </template>
```
Comment (tajespasarela) [inline packages/x-components/src/views/Layout.vue:219]: add NQs content:
```


                    <template #NextQueriesGroup="{ item: { nextQueries } }">
                      <BaseSuggestions #default="{ suggestion }" :suggestions="nextQueries">
                        <NextQuery :suggestion="suggestion" />
                      </BaseSuggestions>
                    </template>
```

File: packages/x-components/src/x-modules/search/components/__tests__/promoteds-list.spec.ts [hunk 7]
```diff
@@ -88,8 +88,8 @@ describe('testing PromotedsList component', () => {
     const { wrapper, getPromoteds } = renderPromotedsList({
       template: `
         <PromotedsList>
-          <template #promoted="{ searchItem }">
-            <p data-test="promoted-slot-overridden">Custom promoted: {{ searchItem.title }}</p>
+          <template #promoted="{ item }">
+            <p data-test="promoted-slot-overridden">Custom promoted: {{ item.title }}</p>
           </template>
         </PromotedsList>`
     });
```
Comment (tajespasarela) [inline packages/x-components/src/x-modules/search/components/__tests__/promoteds-list.spec.ts:95]: rename `x-search-items-list` to `x-items-list` in line 97

File: packages/x-components/src/x-modules/search/components/__tests__/banners-list.spec.ts [hunk 8]
```diff
@@ -88,8 +88,8 @@ describe('testing BannersList component', () => {
     const { wrapper, getBanners } = renderBannersList({
       template: `
         <BannersList>
-          <template #banner="{ searchItem }">
-            <p data-test="banner-slot-overridden">Custom banner: {{ searchItem.title }}</p>
+          <template #banner="{ item }">
+            <p data-test="banner-slot-overridden">Custom banner: {{ item.title }}</p>
           </template>
         </BannersList>`
     });
```
Comment (tajespasarela) [inline packages/x-components/src/x-modules/search/components/__tests__/banners-list.spec.ts:95]: rename `x-search-items-list` to `x-items-list` in line 97

File: packages/x-components/src/x-modules/search/components/__tests__/results-list.spec.ts [hunk 9]
```diff
@@ -88,8 +88,8 @@ describe('testing Results list component', () => {
     const { wrapper, getResults } = renderResultsList({
       template: `
         <ResultsList>
-          <template #result="{ searchItem }">
-            <p data-test="result-slot-overridden">Custom result: {{ searchItem.name }}</p>
+          <template #result="{ item }">
+            <p data-test="result-slot-overridden">Custom result: {{ item.name }}</p>
           </template>
         </ResultsList>`
     });
```
Comment (tajespasarela) [inline packages/x-components/src/x-modules/search/components/__tests__/results-list.spec.ts:95]: rename `x-search-items-list` to `x-items-list` in line 97

File: packages/x-components/src/x-modules/search/components/promoteds-list.vue [hunk 10]
```diff
@@ -6,11 +6,11 @@
         @binding {Vue | string} animation - Animation to animate the elements.
```
Comment (tajespasarela) [inline packages/x-components/src/x-modules/search/components/promoteds-list.vue:6]: change `injected search items to render.` to `injected list items to render.` in line 5

File: packages/x-components/src/x-modules/next-queries/components/next-queries-list.vue [hunk 11]
```diff
@@ -0,0 +1,278 @@
...
+    @Prop({ default: 24 })
+    public frequency!: number;
+
+    /**
+     * The maximum amount of next queries to add in a single group.
+     *
+     * @public
+     */
+    @Prop({ default: 4 })
+    public maxNextQueriesPerGroup!: number;
```
Comment (tajespasarela) [inline packages/x-components/src/x-modules/next-queries/components/next-queries-list.vue:85]: `maxItemsToRender` to keep naming consistency?

**Review by tajespasarela — CHANGES_REQUESTED** [review]

File: packages/x-components/src/components/items-list.vue [hunk 12]
```diff
@@ -1,41 +1,41 @@
...
+      v-for="item in computedItems"
+      :key="item.id"
       class="x-search-items-list__item"
-      :class="searchItem.class"
-      :data-test="searchItem.dataTest"
+      :class="item.class"
+      :data-test="item.dataTest"
     >
       <!--
         @slot Customized Search items list search item.
```
[the remark this answers is not in the document]
Comment (javieri-empathy) [reply packages/x-components/src/components/items-list.vue:17]: Custom item is enough for me. We don't have to repeat the full context each time.

File: packages/x-components/src/x-modules/next-queries/components/next-queries-list.vue [hunk 13]
```diff
@@ -0,0 +1,278 @@
...
+  import { xComponentMixin } from '../../../components/x-component.mixin';
+  import { groupItemsBy, SearchItem } from '../../../utils';
+  import ResultsList from '../../search/components/results-list.vue';
+  import { nextQueriesXModule } from '../x-module';
+
+  /**
+   * A list of next queries.
+   */
+  interface NextQueriesGroup extends SearchItem {
+    modelName: 'NextQueriesGroup';
```
[the remark this answers is not in the document]
Comment (javieri-empathy) [reply packages/x-components/src/x-modules/next-queries/components/next-queries-list.vue:36]: It is used to create slots with that name. You can see so in the `ItemsList` component, or in one of the grid components.

File: packages/x-components/src/x-modules/next-queries/components/next-queries-list.vue (hunk 11 above)
Comment (javieri-empathy) [reply packages/x-components/src/x-modules/next-queries/components/next-queries-list.vue:85]: I think that's  confusing for this prop, here you are talking about a list of lists, and `maxItemsToRender` does not explicitly say if you are talking about the amount of groups or about the amount of next queries to render in each group.

**Review by tajespasarela — APPROVED** [review]

**Pull request merged**